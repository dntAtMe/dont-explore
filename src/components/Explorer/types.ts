import { NodeData } from "../../types";

export interface ExplorerProps {
	structure: NodeData[];
}