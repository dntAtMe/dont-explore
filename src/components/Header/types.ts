export type HeaderProps = {
  onClick: () => void;
  onSearch: (_: string) => void;
};