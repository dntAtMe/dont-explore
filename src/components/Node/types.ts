import { NodeData } from "../../types";

export type NodeProps = {
  data: NodeData;
  onNodeClick: any;
};