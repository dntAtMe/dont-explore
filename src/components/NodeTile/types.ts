import { Node } from "../../utils/Node/Node";

export type NodeTileProps = {
  data: Node;
  onNodeClick: any;
};