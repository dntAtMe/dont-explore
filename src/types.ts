export type NodeData = {
  name: string,
  mode: string,
  size: number,
  modification_date: number,
};
