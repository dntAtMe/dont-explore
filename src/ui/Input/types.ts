export type InputProps = {
  validate?: (value: string) => boolean;
}