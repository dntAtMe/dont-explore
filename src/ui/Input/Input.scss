@import '../../App.scss';

input {
  &.valid {
    border: 1px solid $color-success;
  }
  &.invalid {
    border: 1px solid $color-failure;
  }
  &.neutral {
    border: 1px solid $color-neutral;
  }

  &:focus {
    outline: none;
  }
}