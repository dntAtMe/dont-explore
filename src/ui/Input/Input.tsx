import { useState } from 'react';
import { InputProps } from "./types";

import './Input.scss';

export const Input = ({ validate }: InputProps) => {

  const [isValid, setIsValid] = useState<boolean | undefined>(undefined);

  const onInput = (e: any) => {
    if (validate)
      setIsValid(validate(e.target.value))
  }

  const getValidationClassName = () => {
    switch (isValid) {
      case true:
        return 'valid';
      case false:
        return 'invalid';
      default:
        return 'neutral';
    }
  }

  return (
    <input className={`${getValidationClassName()}`} type="text" onInput={onInput} />
  )
};

export default Input;