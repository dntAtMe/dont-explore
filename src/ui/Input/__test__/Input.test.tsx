import { describe, it } from "@jest/globals";
import {fireEvent, render, screen} from '@testing-library/react'
import Input from "../Input";
import { InputProps } from '../types';

describe('Input w/ validation', () => {
  const inputPropsOk: InputProps = {
    validate: jest.fn().mockReturnValue(true)
  };

  const inputPropsFail: InputProps = {
    validate: jest.fn().mockReturnValue(false)
  };

  it("should render properly when validation successful", () => {
    const rendered = render(<Input validate={inputPropsOk.validate} />)
    const input = rendered.getByTestId('test-input');

    fireEvent.input(input, { target: { value: 'test' } });

    expect(inputPropsOk.validate).toBeCalledTimes(1);
    expect(inputPropsOk.validate).toBeCalledWith('test');
    expect(rendered.asFragment()).toMatchSnapshot();
  });

  it("should render properly when validation failed", () => {
    const rendered = render(<Input validate={inputPropsFail.validate} />)
    const input = rendered.getByTestId('test-input');

    fireEvent.input(input, { target: { value: 'test' } });

    expect(inputPropsFail.validate).toBeCalledTimes(1);
    expect(inputPropsFail.validate).toBeCalledWith('test');
    expect(rendered.asFragment()).toMatchSnapshot();
  });
});

describe('Input w/out validation', () => {
  const inputProps: InputProps = { } as unknown as InputProps;

  it("should render properly", () => {
    const { asFragment } = render(<Input />)
    expect(asFragment()).toMatchSnapshot();
  });
});