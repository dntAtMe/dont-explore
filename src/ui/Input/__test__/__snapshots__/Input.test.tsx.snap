// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input w/ validation should render properly when validation failed 1`] = `
<DocumentFragment>
  <input
    class="invalid"
    data-testid="test-input"
    type="text"
  />
</DocumentFragment>
`;

exports[`Input w/ validation should render properly when validation successful 1`] = `
<DocumentFragment>
  <input
    class="valid"
    data-testid="test-input"
    type="text"
  />
</DocumentFragment>
`;

exports[`Input w/out validation should render properly 1`] = `
<DocumentFragment>
  <input
    class="neutral"
    data-testid="test-input"
    type="text"
  />
</DocumentFragment>
`;
